langchain/src/callbacks/manager.ts

```diff
@@ -395,8 +395,15 @@ export class CallbackManager
     inherit = true
   ): CallbackManager {
     const manager = new CallbackManager(this._parentRunId);
-    manager.setHandlers([...this.handlers].map((handler) => handler.copy()));
+    for (const handler of this.handlers) {
+      const inheritable = this.inheritableHandlers.includes(handler);
+      const copied = handler.copy();
+      manager.addHandler(copied, inheritable);
+    }
     for (const handler of additionalHandlers) {
+      if (manager.handlers.some((h) => h.name === handler.name)) {
```

Comment:
This is to ensure we don't end up with multiple copies of the same handler in here, if eg someone sets it both in constructor and `call()`

Reply:
I see the reason for this but what if someone has an actual usecase where they use the same callback handler (say a websockethandler) two different ways? Let's say there is one they pass through that writes to one websocket but one on the object itself that is writes to a different one. In this case, the one that was passed through will be used while the other one on the object won't be used, potentially leading to unexpected behavior.